lib/synapse/config_generator/haproxy.rb

```diff
@@ -961,17 +980,29 @@ def validate_haproxy_stanza(stanza, stanza_type, service_name)
 
     # generates an individual stanza for a particular watcher
     def generate_frontend_stanza(watcher, config)
-      unless watcher.haproxy.has_key?("port")
+      watcher_config = watcher.generator_config[name]
+      unless watcher_config.has_key?("port")
         log.debug "synapse: not generating frontend stanza for watcher #{watcher.name} because it has no port defined"
         return []
+      else
+        port = watcher_config['port']
       end
 
-      bo = watcher.haproxy['bind_options']
+      bind_options = watcher_config['bind_options']
+      bind_options = bind_options.nil? ? '' : ' ' + bind_options
```

Comment:
Sure, done. I can also fixup the address:port logic in the nginx review as well to do a similar thing.